Sources/GRPC/AsyncAwaitSupport/GRPCAsyncServerCallContext.swift

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Logging
+import NIOHPACK
+
+#if compiler(>=5.5)
+
+/// A context provided to RPC handlers.
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public protocol GRPCAsyncServerCallContext {
+  /// Request headers for this request.
+  var headers: HPACKHeaders { get }
```

Comment:
> We'll need a way to set response headers too (we don't currently support this but we should!)

Do you mean it isn't even supported in the existing, non-async-await, handlers? I took a look and I don't see it; I just see (request) `headers` and (response) `trailers`.

I could play around with what you suggest.

Reply:
> Do you mean it isn't even supported in the existing, non-async-await, handlers?

That's right (although there have been requests to add it). We don't have to add this now but we should leave ourselves enough room in the API that we can add it without the naming being ambiguous.

Reply:
> I could play around with what you suggest.

Up to you; play around and see what feels most natural. 

Side note: I wonder also if we should use "metadata" rather than "headers" and  "{initial,trailing}metadata" rather than response headers and response trailers to align more with the client API.

Reply:
> Side note: I wonder also if we should use "metadata" rather than "headers" and "{initial,trailing}metadata" rather than response headers and response trailers to align more with the client API.

That would be my thinking if we decide to add this.